### PR TITLE
Integrate correctness testing

### PR DESCRIPTION
**Pull request by minitu**
Tom pointed me to https://github.com/N-BodyShop/tests/tree/master/forces/np for correctness testing, which uses `lambs.00200_subsamp_30K` as the input file (30K particles).
The particle accelerations can be output into a file and compared with the `direct.acc` file, which can be done via `tstforce_ChaNGa.sh script`.
This PR adds functionality to output particle accelerations into a file.

### CODE REVIEW

**Comment by minitu** [comment]
Currently outputs all zeros. Need to integrate ChaNGa's `initialForces` which should be run to compute initial accelerations before printing.

**Comment by minitu** [comment]
You can now just run `tests/acc_test.sh` to have the script run ParaTreeT with the 30K subsampled lambs input file and compare the accelerations.

Current error values with `theta = 0.7` and `gconst = 6.6742`:
```
RMS relative force error:
991.25732808810812
Maximum relative force error:
122708.30097939254
```

It may be that our values are way off because we are not using multipole expansion (hexadecapole, as mentioned in the original test script). Will need to ask Tom about this.

**Comment by minitu** [comment]
Despite the current error values, I think the correctness test itself is ready to be merged in.
Don't worry about the file additions in the `array` folder, they were copied over from https://github.com/N-BodyShop/tests/tree/master/array.

File: src/TreePiece.h [hunk 1]
```diff
@@ -566,4 +569,47 @@ void TreePiece<Data>::print(Node<Data>* node) {
...
+      Particle& particle = leaf->particles[i];
+      Real outval;
+      if (dim_cnt == 0) outval = particle.acceleration.x;
+      else if (dim_cnt == 1) outval = particle.acceleration.y;
+      else if (dim_cnt == 2) outval = particle.acceleration.z;
+      fprintf(fp, "%.14g\n", outval);
+    }
+  }
+
+  if (++dim_cnt == 3) dim_cnt = 0;
```
Comment (Wingpad) [inline src/TreePiece.h:597]: Might consider writing this as `dim_cnt = (dim_cnt + 1) % 3`

Comment (minitu) [reply src/TreePiece.h:597]: Good point

**Pull request merged**